data-prepper-plugins/cloudwatch-logs/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/cloudwatch_logs/CouldWatchLogsIT.java

```diff
@@ -0,0 +1,276 @@
+
+    }
+
+    @Test
+    void TestSinkOperationWithBatchSize() throws Exception {
+        long startTime = Instant.now().toEpochMilli();
+        when(thresholdConfig.getBatchSize()).thenReturn(1);
+        when(thresholdConfig.getMaxEventSizeBytes()).thenReturn(1000L);
+        when(thresholdConfig.getMaxRequestSizeBytes()).thenReturn(1000L);
+        
+        sink = createObjectUnderTest();
+        Collection<Record<Event>> records = getRecordList(2);
+        sink.doOutput(records);
+        await().atMost(Duration.ofSeconds(30))
+                .untilAsserted(() -> {
+                    long endTime = Instant.now().toEpochMilli();
+                    GetLogEventsRequest getRequest = GetLogEventsRequest
+                                       .builder()
+                                       .logGroupName(logGroupName)
+                                       .logStreamName(logStreamName)
+                                       .startTime(startTime)
+                                       .endTime(endTime)
+                                       .build();
+                    GetLogEventsResponse response = cloudWatchLogsClient.getLogEvents(getRequest);
+                    List<OutputLogEvent> events = response.events();
+                    assertThat(events.size(), equalTo(2));
+                    for (int i = 0; i < events.size(); i++) {
+                        String message = events.get(i).message();
+                        Map<String, Object> event = objectMapper.readValue(message, Map.class);
+                        assertThat(event.get("name"), equalTo("Person"+i));
+                        assertThat(event.get("age"), equalTo(Integer.toString(i)));
+                    }
+                });
+
+    }
+
+    @Test
+    void TestSinkOperationWithMaxRequestSize() throws Exception {
+        long startTime = Instant.now().toEpochMilli();
+        when(thresholdConfig.getBatchSize()).thenReturn(20);
+        when(thresholdConfig.getMaxRequestSizeBytes()).thenReturn(108L);
+        
+        sink = createObjectUnderTest();
+        Collection<Record<Event>> records = getRecordList(2);
+        sink.doOutput(records);
+        await().atMost(Duration.ofSeconds(30))
+                .untilAsserted(() -> {
+                    long endTime = Instant.now().toEpochMilli();
+                    GetLogEventsRequest getRequest = GetLogEventsRequest
+                                       .builder()
+                                       .logGroupName(logGroupName)
+                                       .logStreamName(logStreamName)
+                                       .startTime(startTime)
+                                       .endTime(endTime)
+                                       .build();
+                    GetLogEventsResponse response = cloudWatchLogsClient.getLogEvents(getRequest);
+                    List<OutputLogEvent> events = response.events();
+                    assertThat(events.size(), equalTo(2));
```

Comment:
Should this be `greaterThanOrEqualTo()`? If multiple tests run we may find multiple records. It seems ideal to be lenient on this considering you have other validations below.

Reply:
The logstream is created dynamically and removed. Even if multiple of them run, it shouldn't cause any issues. Also, there is no workflow to run this yet.